<?xml version="1.0" encoding="UTF-8"?>
<launch>

	<arg name="only_use_rf2o" default="false" />

	<arg name="pub_tf" default="false"/>

	<arg name="odom_used" default="true"/>
	<arg name="imu_used" default="false"/>
	<arg name="vo_used" default="false"/>

	<param name="output_frame" value="odom_combined"/>

	<!-- base node -->
	<node pkg="lemon_minibot_control" name="base_control"  type="base.py" output="screen">
		<!-- some times, the encoder will not correct, we can scale it -->
		<param name="scale"   value="2"/> 
		<param name="pub_tf"   value="$(arg pub_tf)"/>   
		<param name="scale"   value="2"/>   
	</node>

	<!-- star ydlidar  -->
	<include file="$(find lemon_minibot_control)/launch/ydlidar.launch"/>

	<!-- rf2o -->
	<include file="$(find lemon_minibot_control)/launch/rf2o.launch"/>

	<include file="$(find lemon_minibot_control)/launch/tf.launch"/> 

	<!-- robot_pose_ekf -->
	<!--<include file="$(find lemon_minibot_control)/launch/rf2o.launch"/> 
		<arg name="odom_used" value="$(arg odom_used)"/>
		<arg name="imu_used" value="$(arg imu_used)"/>
		<arg name="vo_used" value="$(arg vo_used)"/>

  		<remap from="imu_data" to="imu" />	
	</include> -->
	<!-- default subscribe /imu_data  /odom  /vo-->

	<!-- open gmapping -->
	<!-- <include file="$(find lemon_minibot_core)/launch/gmapping.launch">
		<arg name="odom_frame" value="odom"/> 
		
	</include> -->

	<!-- <group if="$(arg only_use_rf2o)">
		<param name="pub_tf" value="false"/>
	</group>
	<group unless="$(arg only_use_rf2o)">
		<node pkg="lemon_minibot" name="keyboard_teleop" type="keyboard_teleop.py" output="screen"/>
	</group> -->
	

</launch>
